plugins/node/instrumentation-mongoose/src/utils.ts

```diff
@@ -20,8 +20,10 @@ import { safeExecuteInTheMiddle } from '@opentelemetry/instrumentation';
 import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
 
 export function getAttributesFromCollection(
-  collection: Collection
+  _collection: unknown
```

Comment:
I might be missing something, but why is this function part of the public api? it is not exported in `index.ts`

Reply:
It is not exported to `index.ts`, but it's exported from utils. I expected TS to recursively import ts files and compile them. It seems that I was wrong and in the normal case utils is not compiled. Only when 

```
import * as mongooseUtils from '@opentelemetry/instrumentation-mongoose/build/src/utils';
```

is run explicitly, the problem occurs. Is there a compiler option that turns that process recursive?

That said, if I understand the situation correctly, we don't have to remove the type from that signature. Thanks for pointing that out, @blumamir!

Reply:
> Is there a compiler option that turns that process recursive?

From what I know, typescript is compiling only the single `.d.ts` file that is declared in `package.json` like this:
```js
{
  "types": "index.d.ts"
}
```

This file in turn can `import` other type files (or not), but since `utils.d.ts` is not imported from `index.d.ts`, then typescript never encounters `import * from 'mongoose'` so there are no issues.
This is only my understanding, but I could be missing some important details. typescript is so complicated 😅